keras_hub/src/models/gemma3/gemma3_backbone.py

```diff
@@ -0,0 +1,382 @@
+            in all transformer blocks. Defaults to `1e-6`.
+        dropout: float. Dropout probability for the Transformer decoder blocks.
+            Defaults to `0`.
+        dtype: string or `keras.mixed_precision.DTypePolicy`. The dtype to use
+            for the models computations and weights. Note that some
+            computations, such as softmax and layer normalization will always
+            be done a float32 precision regardless of dtype.
+
+    Example:
+    ```python
+    input_data = {
+        "token_ids": np.ones(shape=(1, 12), dtype="int32"),
+        "images": np.random.uniform(size=(1, 224, 224, 3)),
+        "padding_mask": np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]]),
+    }
+
+    # Pretrained Gemma3 decoder.
+    model = keras_hub.models.Gemma3Backbone.from_preset("gemma3")
+    model(input_data)
+
+    # Randomly initialized Gemma3 decoder with custom config.
+    model = keras_hub.models.Gemma3Backbone(
+        vocabulary_size=50257,
+        images_size=224,
+        num_layers=12,
+        num_query_heads=12,
+        num_key_value_heads=1,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        head_dim=64,
+        vit_patch_size=14,
+        vit_num_heads=8,
+        vit_hidden_dim=768,
+        vit_intermediate_dim=3072,
+        vit_num_layers=2,
+    )
+    model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        image_size,
+        image_max_length,
+        num_layers,
+        num_query_heads,
+        num_key_value_heads,
+        hidden_dim,
+        intermediate_dim,
+        head_dim,
+        vit_patch_size,
+        vit_num_heads,
+        vit_hidden_dim,
+        vit_num_layers,
+        vit_intermediate_dim=None,  # TODO remove default
+        vit_pooling=None,
+        vit_pool_size=None,
+        vit_name=None,
+        query_head_dim_normalize=True,
+        use_qk_norm=True,
+        use_post_ffw_norm=False,
+        use_post_attention_norm=False,
+        attention_logit_soft_cap=None,
+        final_logit_soft_cap=None,
+        use_sliding_window_attention=False,
+        sliding_window_size=4096,
+        layer_norm_epsilon=1e-6,
+        dropout=0,
+        dtype=None,
+        **kwargs,
+    ):
+        # === Layers ===
+        self.token_embedding = ReversibleEmbedding(
+            input_dim=vocabulary_size,
+            output_dim=hidden_dim,
+            tie_weights=True,
+            embeddings_initializer=keras.initializers.VarianceScaling(
+                scale=1.0,
+                mode="fan_in",
+                distribution="untruncated_normal",
+                seed=None,
+            ),
+            dtype=dtype,
+            logit_soft_cap=final_logit_soft_cap,
+            name="token_embedding",
+        )
+
+        # TODO Remove this. Work around for previous serialization bug.
+        vit_intermediate_dim = vit_intermediate_dim or 4304
```

Comment:
what's the deal with this?

Reply:
Probably we accidentally serialized some models with the intermediate dim in the config. We have no legacy configs for gemma3 right now let's definitely ditch.